### ISSUE TITLE
Ingredient manifests are being regenerated for hashing which can differ from what the file contains

### ISSUE THREAD

**Issue by DavidS-msft** [contentauth/c2pa-rs#504]
While testing we discovered that the c2patool would flag a bad hash for an ingredient's manifest. After adding code to dump out the manifest that the tool was hashing we noticed that in the data stored in the databoxes the tool was adding a 3rd field "data_types" with the simple item value of 22 which is not in the original file. According to the c2pa spec the "data_types" field is optional.

Digging in the function calc_manifest_box_hash builds the store from the claim and hashes the bytes. The issue is that the generated store doesn't match the store from the file, specifically around the data in the databoxes.

Attaching a zip with the ingredient manifests and the source image to help with debugging.

[No_data_types_manifests.zip](https://github.com/user-attachments/files/16200620/No_data_types_manifests.zip)
Test Image: https://1drv.ms/i/s!Ajb0eiGfJqsdosxeN042ifu1KgbOow?e=4tY1bP

**Comment by mauricefisher64** [comment]
Good catch, the data_types field should be optional when we serialize.  Pushing fix shortly.

**Comment by mauricefisher64** [comment]
fixed in  PR: https://github.com/contentauth/c2pa-rs/pull/512
